### PR TITLE
feat(catalog): Incorporate catalog feedback

### PR DESCRIPTION
**Pull request by lordrip**
### Context
After merging https://github.com/KaotoIO/kaoto-next/pull/10, there were still improvements pending. This pull request brings the first batch of those improvements.

### Notes
Please see @apupier [comments here](https://github.com/KaotoIO/kaoto-next/pull/10#pullrequestreview-1573126045) for more details.

### Pending changes
* Add more tests for the rest of the components
* Enable the search to ignore accents and casing

### Changes

* Replace `Tabs` with `ToggleGroup` for performance reasons. It turns out that all tabs are always present in the DOM regardless of their visibility, making a big impact on the render performance of the catalog.
image

* The `SearchBar` checks the component's tags.
image

* Update the README.md file with instructions on how to build the project.

* Added tests for the `Camel components to Tile` adapters

### CODE REVIEW

**Review by igarashitm — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,27 @@
...
+This project leverages `vite`, you can find more information about it [here](https://vitejs.dev/config/)
+
+### Requirements
+- NodeJS (>v18.x)
+- yarn (>v3.x)
+
+### Steps
+1. Clone the repository
+2. In the recently cloned folder, run `yarn install` to install all the dependencies
+(By default, `@kaoto-next/camel-catalog` will be built as well, using the `mvn` wrapper)
```
Comment (apupier) [inline README.md:14]: it doesn't seem to trigger it for me:
image

File: README.md [hunk 2]
```diff
@@ -1,2 +1,27 @@
...
+
+#### Running the web application
+Run `yarn workspace @kaoto-next/ui run start` to start the development server. The application will be available at `http://localhost:5173` by default.
```
Comment (apupier) [inline README.md:17]: I ahve an erro when trying to access the Catalog:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at BaseCatalog (BaseCatalog.tsx:22:30)
    at renderWithHooks (react-dom.development.js:16305:18)
    at updateFunctionComponent (react-dom.development.js:19588:20)
    at beginWork (react-dom.development.js:21601:16)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:4164:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:16)
    at invokeGuardedCallback (react-dom.development.js:4277:31)
    at beginWork$1 (react-dom.development.js:27451:7)
    at performUnitOfWork (react-dom.development.js:26557:12)
    at workLoopSync (react-dom.development.js:26466:5)
 ```

Comment (apupier) [reply README.md:17]: After the rebase you did and retrying it is now working

**Review by apupier — APPROVED** [review]

**Review by apupier — CHANGES_REQUESTED** [review]
- it is mentioned that it can search on description but it not working for `Cloudtrail` for instance

https://github.com/KaotoIO/kaoto-next/assets/1105127/c8347ef3-da7e-4527-aacc-13c421be6e92

**Comment by lordrip** [comment]
@apupier I fixed what was missing regarding the automatic build of `@kaoto-next/camel-catalog`

### Freshly cloned repository
Notice that there's no `dist` folder
image

### After `yarn install`
Notice the new `dist` folder
image

### `Yarn` logs
image

File: packages/ui/src/components/Catalog/CatalogFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  }, []);
+
+  return (
+    <Form className={props.className}>
+      <Grid hasGutter md={6}>
+        <GridItem>
+          <FormGroup label="Search term" fieldId="search-term">
+            <SearchInput
+              aria-label="Find by name, description or tag"
+              placeholder="Find by name, description or tag"
```
Comment (apupier) [inline packages/ui/src/components/Catalog/CatalogFilter.tsx:27]: still description mentioned here?

File: packages/ui/src/components/Catalog/CatalogFilter.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+              onChange={props.onChange}
+              onClear={props.onChange}
+              ref={inputRef}
+            />
+          </FormGroup>
+        </GridItem>
+
+        <GridItem className="pf-v5-u-text-align-right">
+          <FormGroup label="Type" fieldId="element-type">
+            <ToggleGroup aria-label="Select erlement type">
```
Comment (apupier) [inline packages/ui/src/components/Catalog/CatalogFilter.tsx:38]: typo `erlement`

File: packages/ui/src/components/Catalog/CatalogFilter.tsx (hunk 3 above)
Comment (lordrip) [reply packages/ui/src/components/Catalog/CatalogFilter.tsx:27]: I forgot to push the changes :face_in_clouds:

File: packages/ui/src/components/Catalog/CatalogFilter.tsx (hunk 4 above)
Comment (lordrip) [reply packages/ui/src/components/Catalog/CatalogFilter.tsx:38]: Just in time to fix this as well

**Review by apupier — APPROVED** [review]
another feedback for later, I think we can remove the `Search term` text:
image

**Pull request merged**